src/message_bus.zig

```diff
@@ -738,14 +738,7 @@ fn MessageBusImpl(comptime process_type: ProcessType) type {
                 // `references` and `header` metadata.
                 if (connection.recv_progress == header.size) return connection.recv_message.?.ref();
 
-                const message = bus.get_message() orelse {
-                    // TODO Decrease the probability of this happening by:
-                    // 1. using a header-only message if possible.
-                    // 2. determining a true upper limit for static allocation.
-                    log.err("no free buffer available to deliver message from {}", .{connection.peer});
-                    connection.terminate(bus, .shutdown);
-                    return null;
-                };
+                const message = bus.get_message();
```

Comment:
:+1: good catch. I did account for one of those in the client, but hadn't considered it more generally.

I think it's enough to just add `+ 1` to the `messages_max_replica` (the client already has it) to handle _all_ the bursts. That is, bursts can never overlap, because the burst message will either be stored in the respective queue/pipeline/etc, or dropped if the respective queue/... is full.

Reply:
Good, and it's also cheaper and more accurate to add the `+ 1` to the total, rather than for each connection, since it only happens one-by-one in the control plane.

Reply:
Yes, exactly!